### PR TITLE
Fix Warnings: incorrect conversion between integer type

### LINKED ISSUE #839: Security:Incorrect conversion between integer types
**What happened**:

**What you expected to happen**:

Fix the security problems as below:

image

image

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by kedadiannao220**
**What this PR does**:

 it is just a warning about [CWE-681](https://github.com/github/codeql-go/blob/main/ql/src/Security/CWE-681/IncorrectIntegerConversion.ql)

**Which issue(s) this PR fixes**:

Fixes #839

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: common/url.go [hunk 1]
```diff
@@ -484,22 +484,49 @@ func (c URL) GetParamBool(key string, d bool) bool {
...
 		return d
 	}
-	return int64(r)
+	return r
+}
+
+// GetParamInt32 gets int32 value by @key
+func (c URL) GetParamInt32(key string, d int32) int32 {
+	r, err := strconv.ParseInt(c.GetParam(key, ""), 10, 32)
+	if r == 0 || err != nil {
```
Comment (wongoo) [inline common/url.go:499]: if `key=0`, then return default value. is it the right logic?

File: common/url.go [hunk 2]
```diff
@@ -484,22 +484,49 @@ func (c URL) GetParamBool(key string, d bool) bool {
...
+		return d
+	}
+	return int32(r)
+}
+
+// GetParamByInt0 gets int value by @key
+func (c URL) GetParamByInt0(key string, d int) int {
```
Comment (wongoo) [inline common/url.go:506]: is it better to rename it to `GetParamByIntValue` ?

File: common/url.go [hunk 3]
```diff
@@ -484,22 +484,49 @@ func (c URL) GetParamBool(key string, d bool) bool {
...
-	r, err := strconv.Atoi(c.GetParam("methods."+method+"."+key, ""))
+	r, err := strconv.ParseInt(c.GetParam("methods."+method+"."+key, ""), 10, 64)
+	if r == 0 || err != nil {
+		return d
+	}
+	return r
+}
+
+// GetMethodParamInt0 gets int method param
+func (c URL) GetMethodParamInt0(method string, key string, d int) int {
```
Comment (wongoo) [inline common/url.go:524]: is it better to rename it to `GetMethodParamIntValue` ?

File: common/url.go [hunk 4]
```diff
@@ -484,22 +484,49 @@ func (c URL) GetParamBool(key string, d bool) bool {
...
+	if r == 0 || err != nil {
+		return d
+	}
+	return int32(r)
+}
+
+// GetParamByInt0 gets int value by @key
+func (c URL) GetParamByInt0(key string, d int) int {
+	r, err := strconv.ParseInt(c.GetParam(key, ""), 10, 0)
+	if r == 0 || err != nil {
```
Comment (zouyx) [inline common/url.go:508]: as wongoo's comment

File: common/url.go [hunk 5]
```diff
@@ -484,22 +484,49 @@ func (c URL) GetParamBool(key string, d bool) bool {
...
+		return d
+	}
+	return int(r)
 }
 
 // GetMethodParamInt gets int method param
 func (c URL) GetMethodParamInt(method string, key string, d int64) int64 {
-	r, err := strconv.Atoi(c.GetParam("methods."+method+"."+key, ""))
+	r, err := strconv.ParseInt(c.GetParam("methods."+method+"."+key, ""), 10, 64)
+	if r == 0 || err != nil {
```
Comment (zouyx) [inline common/url.go:517]: as wongoo's comment

File: common/url.go (hunk 1 above)
Comment (kedadiannao220) [reply common/url.go:499]: `key` is only a flag of `param` value;  this conversion  is about value of  key, not key itself;

e.g.
```
func TestURLGetParamInt(t *testing.T) {
	params := url.Values{}
	params.Set("key", "3")
	u := URL{baseUrl: baseUrl{params: params}}
	v := u.GetParamInt("key", 1)
	assert.Equal(t, int64(3), v)

	u = URL{}
	v = u.GetParamInt("key", 1)
	assert.Equal(t, int64(1), v)
}
``

File: common/url.go (hunk 4 above)
Comment (kedadiannao220) [reply common/url.go:508]: ok , i will change it;

File: common/url.go (hunk 1 above)
Comment (wongoo) [reply common/url.go:499]: maybe u miss-understand what I mean,  if `params.Set("key", "0")` ,  the value of `u.GetParamInt("key", 1)` will be 1, which is not correct.

Comment (kedadiannao220) [reply common/url.go:499]: wow, soga;  you are right.....  I think I am fixing a bug  of this module.

Comment (kedadiannao220) [reply common/url.go:499]: I searched the history commits,  `r==0` is  avoid to situation of  `param.Set("key", "")`

**Review by wongoo — APPROVED** [review]

**Review by zouyx — COMMENTED** [review]
Change target branch to develop pls

**Review by AlexStocks — APPROVED** [review]

**Comment by watermelo** [comment]
Merged at pr #871

**Pull request closed without merging**